api/konnect/v1alpha1/konnect_extension_types.go

```diff
@@ -0,0 +1,167 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	commonv1alpha1 "github.com/kong/kubernetes-configuration/api/common/v1alpha1"
+)
+
+func init() {
+	SchemeBuilder.Register(&KonnectExtension{}, &KonnectExtensionList{})
+}
+
+const (
+	// KonnectExtensionKind holds the kind for the KonnectExtension.
+	KonnectExtensionKind = "KonnectExtension"
+)
+
+// KonnectExtension is the Schema for the KonnectExtension API, and is intended to be referenced as
+// extension by the DataPlane, ControlPlane or GatewayConfiguration APIs.
+// If one of the above mentioned resources successfully refers a KonnectExtension, the underlying
+// deployment(s) spec gets customized to include the konnect-related configuration.
+//
+// +genclient
+// +kubebuilder:resource:scope=Namespaced
+// +kubebuilder:object:root=true
+// +kubebuilder:object:generate=true
+// +kubebuilder:subresource:status
+// +kubebuilder:validation:XValidation:rule="oldSelf.spec.controlPlaneRef == self.spec.controlPlaneRef", message="spec.controlPlaneRef is immutable."
+// +kubebuilder:validation:XValidation:rule="self.spec.controlPlaneRef.type == 'konnectID' ? has(self.spec.konnect) : true",message="konnect must be set when ControlPlaneRef is set to KonnectID."
+// +kubebuilder:validation:XValidation:rule="self.spec.controlPlaneRef.type == 'konnectNamespacedRef' ? !has(self.spec.konnect) : true",message="konnect must be unset when ControlPlaneRef is set to konnectNamespacedRef."
+// +apireference:kgo:include
+// +kong:channels=gateway-operator
+type KonnectExtension struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec is the specification of the KonnectExtension resource.
+	Spec KonnectExtensionSpec `json:"spec,omitempty"`
+	// Status is the status of the KonnectExtension resource.
+	Status KonnectExtensionStatus `json:"status,omitempty"`
+}
+
+// KonnectExtensionList contains a list of KonnectExtension.
+//
+// +kubebuilder:object:root=true
+// +apireference:kgo:include
+type KonnectExtensionList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []KonnectExtension `json:"items"`
+}
+
+// KonnectExtensionSpec defines the desired state of KonnectExtension.
+type KonnectExtensionSpec struct {
+	// ControlPlaneRef is a reference to a Konnect ControlPlane this KonnectExtension is associated with.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:XValidation:rule="self.type != 'kic'", message="kic type not supported as controlPlaneRef."
+	ControlPlaneRef *commonv1alpha1.ControlPlaneRef `json:"controlPlaneRef"`
+
+	// DataPlaneClientAuth is the configuration for the client certificate authentication for the DataPlane.
+	// It is required to set up the connection with the Konnect Platform.
```

Comment:
```suggestion
	// It is required to set up the connection with the Konnect Platform.
	//
```

The field itself is optional, yet the comment states that it is required. Can we clarify this?

Reply:
good catch. i've fixed the comment to reflect the actual state of things.